spec/services/payments/lose_dispute_service_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Payments::LoseDisputeService, type: :service do
+  subject(:lose_dispute_service) { described_class.new(payment:) }
+
+  describe "#call" do
+    context "when payment does not exist" do
+      let(:payment) { nil }
+
+      it "returns a failure" do
+        result = lose_dispute_service.call
+
+        aggregate_failures do
+          expect(result).not_to be_success
+
+          expect(result.error).to be_a(BaseService::NotFoundFailure)
+          expect(result.error.resource).to eq("payment")
+        end
+      end
+
+      it "does not enqueue a send webhook job for the invoice" do
+        expect { lose_dispute_service.call }.not_to have_enqueued_job(SendWebhookJob)
+      end
+    end
+
```

Comment:
I miss one context for when the payable does not exist. Is it even possible? the guard clause is there...